### PR TITLE
fix(traits): Mute trait persists even if a mime breaks their vow

### LINKED ISSUE #26014: Characters with the Mute trait can speak if they play as mime and break their vow of silence
## Description

A very obscure issue, but still an issue. Might be worth making sure other traits don't have similar issues if other systems add or remove their components.

**Reproduction**

- Play as a character with the Mute trait.
- Play as a mime or give yourself the MimePowers component.
- Use the action to break your vow of silence.
- Speak.

**Screenshots**

**Additional context**

MimePowers uses EnsureComp to add the Muted component, and removes it when breaking the vow. It probably should check if the character is already mute before adding Muted so it can know not to remove it.

### PR DESCRIPTION
**Pull request by Brandon-Huu**
## About the PR

Made it so if a mime has the mute trait they do not gain the ability to speak when their vow is broken

## Why / Balance

Fixes #26014

## Technical details

This might create a lot of issues because I changed the way traits are applied.

Instead of ensuring a component exists and not changing it if it does, this PR makes it so if a component for a trait already exists it gets overwritten.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by Brandon-Huu** [comment]
Also, I could probably just make it so mimes can't break their vow if they have the mute trait.

Should I do that instead of this? 

Kinda weird that they can break their vow but still not speak.

File: Content.Shared/Speech/Muting/MutedComponent.cs [hunk 1]
```diff
@@ -6,5 +6,11 @@ namespace Content.Shared.Speech.Muting
     public sealed partial class MutedComponent : Component
     {
 
+      /// <summary>
+      /// Whether the entity should be able to be unmuted
+      /// </summary>
+      [DataField]
+      public bool removable = true;
+
     }
```
Comment (metalgearsloth) [inline Content.Shared/Speech/Muting/MutedComponent.cs:15]: This sucks just use an event instead, also it's not even using C# / SS14 conventions for pascal case.

Comment (Brandon-Huu) [reply Content.Shared/Speech/Muting/MutedComponent.cs:15]: Oops

**Comment by Brandon-Huu** [comment]
bruh

**Pull request closed without merging**